### PR TITLE
[Merged by Bors] - chore: rename `StructureGroupoid.eq_on_source'` to `StructureGroupoid.mem_of_eqOnSource'`

### PR DESCRIPTION
**Pull request by winstonyin**
Since it refers to `PartialEquiv.EqOnSource`, the correct naming scheme should not be snake case `eq_on_source`. I also added `mem_of_` because that's the target of the lemma, while `EqOnSource` is just a hypothesis.

There are no added lemmas or docstrings in this PR. It's all just renaming.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/ChartedSpace.lean [hunk 1]
```diff
@@ -250,10 +250,14 @@ theorem StructureGroupoid.locality (G : StructureGroupoid H) {e : PartialHomeomo
...
+theorem StructureGroupoid.mem_of_eqOnSource (G : StructureGroupoid H) {e e' : PartialHomeomorph H H}
     (he : e ∈ G) (h : e' ≈ e) : e' ∈ G :=
-  G.eq_on_source' e e' he h
-#align structure_groupoid.eq_on_source StructureGroupoid.eq_on_source
+  G.mem_of_eqOnSource' e e' he h
+#align structure_groupoid.eq_on_source StructureGroupoid.mem_of_eqOnSource
+
+theorem StructureGroupoid.mem_iff_of_eqOnSource (G : StructureGroupoid H)
+    {e e' : PartialHomeomorph H H} (h : e ≈ e') : e ∈ G ↔ e' ∈ G :=
+  ⟨fun he ↦ G.mem_of_eqOnSource he (Setoid.symm h), fun he' ↦ G.mem_of_eqOnSource he' h⟩
```
Comment (winstonyin) [inline Mathlib/Geometry/Manifold/ChartedSpace.lean:260]: In addition to the rename, I added this one small lemma, an `iff` version of `mem_of_eqOnSource`. This is to mirror how lemmas like `HasDerivAt.congr_of_eventuallyEq` have an `iff` counterpart `Filter.EventuallyEq.hasDerivAt_iff`. If there is any doubt about this, I'll be happy to remove it from this particular PR.

File: scripts/nolints.json [hunk 2]
```diff
@@ -1037,7 +1037,7 @@
  ["docBlame", "StrictWeakOrder.Equiv"],
  ["docBlame", "StrictWeakOrder.«term_≈[_]_»"],
  ["docBlame", "Structomorph.mem_groupoid"],
- ["docBlame", "StructureGroupoid.eq_on_source'"],
+ ["docBlame", "StructureGroupoid.mem_of_eqOnSource'"],
```
Comment (grunweg) [inline scripts/nolints.json:1040]: Can you add a docstring to `mem_of_eqOnSource'` instead? Now that you've thought about this, it shouldn't be too hard, right?

**Review by grunweg — APPROVED** [review]
Makes sense to me. Thanks for doing this!

About splitting the lemma addition into a separate easy PR: this would make reviewing the PR easier, but having these in separate commits would also suffice. (In such case, I put "best reviewed commit-by-commit" into the description. If a comment is auto-generated/mechanical, that can also be mentioned there.)

File: Mathlib/Geometry/Manifold/ChartedSpace.lean (hunk 1 above)
Comment (winstonyin) [reply Mathlib/Geometry/Manifold/ChartedSpace.lean]: This lemma has been removed from this PR and moved to #9827.

File: scripts/nolints.json (hunk 2 above)
Comment (winstonyin) [reply scripts/nolints.json:1040]: Both the added lemma and docstrings are now on #9827.

File: Mathlib/Geometry/Manifold/ChartedSpace.lean [hunk 3]
```diff
@@ -167,7 +167,7 @@ structure StructureGroupoid (H : Type u) [TopologicalSpace H] where
   id_mem' : PartialHomeomorph.refl H ∈ members
   locality' : ∀ e : PartialHomeomorph H H,
     (∀ x ∈ e.source, ∃ s, IsOpen s ∧ x ∈ s ∧ e.restr s ∈ members) → e ∈ members
-  eq_on_source' : ∀ e e' : PartialHomeomorph H H, e ∈ members → e' ≈ e → e' ∈ members
+  mem_of_eqOnSource' : ∀ e e' : PartialHomeomorph H H, e ∈ members → e' ≈ e → e' ∈ members
```
Comment (ocfnash) [inline Mathlib/Geometry/Manifold/ChartedSpace.lean:170]: I see, the point is that the equivalence `≈` appearing here is `PartialEquiv.EqOnSource`.

**Review by ocfnash — APPROVED** [review]
Thanks!

bors merge

**Pull request closed without merging**